tests/out/hlsl/access.hlsl

```diff
@@ -114,10 +125,53 @@ void test_matrix_within_struct_accesses()
     return;
 }
 
+MatCx2InArray ConstructMatCx2InArray(float4x2 arg0[2]) {
+    MatCx2InArray ret = (MatCx2InArray)0;
+    ret.am = arg0;
+    return ret;
+}
```

Comment:
Is this HLSL really correct? Given the code above:

```
struct MatCx2InArray {
    struct { float2 _0; float2 _1; float2 _2; float2 _3; } am[2];
};
```

Isn't line 9 there assigning a `float4x2[2]` to an array of two anonymous structs?

Reply:
No, I don't think so. IIRC implicit casts only work the other way (i.e. structs with same layout as a primitive being casted to said primitive)